### DISCUSSION TITLE
CarlaUE4 make error:"xxx.generated.h" could not find

### DISCUSSION THREAD

**Discussion by lq666666666** [carla-simulator/carla#6662] [General]
1.git clone --depth 1 -b carla https://github.com/CarlaUnreal/UnrealEngine.git ~/UnrealEngine_4.26

2.cd ~/UnrealEngine_4.26

3. ./Setup.sh
`Target arch set to: x86_64-unknown-linux-gnu
Building ThirdParty libraries

If you don't see SUCCESS message in the end, then building did not finish properly.
In that case, take a look into /home/carma/UnrealEngine_4.26/Engine/Build/BatchFiles/Linux/BuildThirdParty.log for details.

No third party libs needed to be built locally

**********  SUCCESS ****************
Setup successful.
`

4../GenerateProjectFiles.sh
`Attempting to set up UE4 pretty printers for gdb (existing UE4Printers.py, if any, will be overwritten)...
	updated UE4Printers.py
	found necessary entries in ~/.gdbinit file, not changing it.

Setting up Unreal Engine 4 project files...

Fixing inconsistent case in filenames.
Setting up Mono
Generating data for project indexing... 100%
Generating data for project indexing... 100%
Writing project files... 100%
Generating data for project indexing... 100%
Generating data for project indexing... 100%
Generating data for project indexing... 100%
Generating data for project indexing... 100%
Writing project files... 100%`

5. make
bash "/home/carma/UnrealEngine_4.26/Engine/Build/BatchFiles/Linux/Build.sh" CrashReportClient Linux Shipping  
Fixing inconsistent case in filenames.
Setting up Mono
Running command : Engine/Binaries/DotNET/UnrealBuildTool.exe CrashReportClient Linux Shipping
Using 'git status' to determine working set for adaptive non-unity build (/home/carma/UnrealEngine_4.26).
Creating makefile for CrashReportClient (no existing makefile)
------- Build details --------
Using toolchain located at '/home/carma/UnrealEngine_4.26/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v17_clang-10.0.1-centos7/x86_64-unknown-linux-gnu'.
Using clang (/home/carma/UnrealEngine_4.26/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v17_clang-10.0.1-centos7/x86_64-unknown-linux-gnu/bin/clang++) version '10.0.1' (string), 10 (major), 0 (minor), 1 (patch)
Using bundled libc++ standard C++ library.
Using lld linker
Using llvm-ar : /home/carma/UnrealEngine_4.26/Engine/Extras/ThirdPartyNotUE/SDKs/HostLinux/Linux_x64/v17_clang-10.0.1-centos7/x86_64-unknown-linux-gnu/bin/llvm-ar
Using fast way to relink  circularly dependent libraries (no FixDeps).
------------------------------
Waiting for 'git status' command to complete
Building CrashReportClient...
Performing 73 actions (16 in parallel)
[8/73] Compile Module.TraceLog.cpp
[5/73] Compile SharedPCH.Slate.ShadowErrors.h
[6/73] Compile PCH.ImageWrapper.h
[1/73] Compile PCH.Core.h
[4/73] Compile PCH.CoreUObject.h
[3/73] Compile SharedPCH.CoreUObject.ShadowErrors.h
[2/73] Compile SharedPCH.Core.ShadowErrors.h
[7/73] Compile BuildSettings.cpp
[9/73] Compile Module.ImageWrapper.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/ImageWrapper/Module.ImageWrapper.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/ImageWrapper/Private/Formats/BmpImageWrapper.cpp:6:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/ImageWrapper/Public/BmpImageSupport.h:8:10: fatal error: 'BmpImageSupport.generated.h' file not found
#include "BmpImageSupport.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[10/73] Compile Module.Core.18_of_18.cpp
[11/73] Compile Module.Core.5_of_18.cpp
[12/73] Compile Module.Core.9_of_18.cpp
[13/73] Compile Module.Core.17_of_18.cpp
[14/73] Compile Module.Core.13_of_18.cpp
[15/73] Compile Module.Core.14_of_18.cpp
[16/73] Compile Module.Core.15_of_18.cpp
[17/73] Compile Module.Core.7_of_18.cpp
[18/73] Compile Module.Core.1_of_18.cpp
[19/73] Compile Module.Core.4_of_18.cpp
[20/73] Compile Module.DesktopPlatform.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/Slate/SharedPCH.Slate.ShadowErrors.h:66:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/Slate/Public/SlateSharedPCH.h:255:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/InputCore/Classes/InputCoreTypes.h:10:10: fatal error: 'InputCoreTypes.generated.h' file not found
#include "InputCoreTypes.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[21/73] Compile Module.Slate.1_of_7.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/Slate/Module.Slate.1_of_7.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/Slate/Private/Framework/Application/AnalogCursor.cpp:4:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/InputCore/Classes/InputCoreTypes.h:10:10: fatal error: 'InputCoreTypes.generated.h' file not found
#include "InputCoreTypes.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
[22/73] Compile Module.DirectoryWatcher.cpp
1 error generated.
[23/73] Compile Analytics.cpp
[24/73] Compile Module.StandaloneRenderer.cpp
1 error generated.
[25/73] Compile Module.Slate.4_of_7.cpp
[26/73] Compile Module.AnalyticsET.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/StandaloneRenderer/Module.StandaloneRenderer.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Developer/StandaloneRenderer/Private/Linux/OpenGL/SlateOpenGLContext.cpp:7:
In file included from Developer/StandaloneRenderer/Private/OpenGL/SlateOpenGLRenderer.h:7:
In file included from Runtime/SlateCore/Public/Layout/SlateRect.h:6:
In file included from Runtime/SlateCore/Public/Layout/Margin.h:7:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Public/Types/SlateEnums.h:8:10: fatal error: 'SlateEnums.generated.h' file not found
#include "SlateEnums.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/Slate/Module.Slate.4_of_7.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/Slate/Private/Framework/Text/SlateTextUnderlineLineHighlighter.cpp:3:
In file included from Runtime/Slate/Public/Framework/Text/SlateTextUnderlineLineHighlighter.h:5:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Public/Styling/SlateColor.h:9:10: fatal error: 'SlateColor.generated.h' file not found
#include "SlateColor.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[27/73] Compile Module.Slate.6_of_7.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/Slate/Module.Slate.6_of_7.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/Slate/Private/Widgets/Layout/SSeparator.cpp:3:
In file included from Runtime/Slate/Public/Widgets/Layout/SSeparator.h:6:
In file included from Runtime/SlateCore/Public/Widgets/DeclarativeSyntaxSupport.h:8:
In file included from Runtime/SlateCore/Public/Layout/Clipping.h:7:
In file included from Runtime/SlateCore/Public/Layout/SlateRect.h:6:
In file included from Runtime/SlateCore/Public/Layout/Margin.h:7:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Public/Types/SlateEnums.h:8:10: fatal error: 'SlateEnums.generated.h' file not found
#include "SlateEnums.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~
[28/73] Compile RSA.cpp
[29/73] Compile Module.XmlParser.cpp
1 error generated.
[30/73] Compile Module.Slate.5_of_7.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/Slate/Module.Slate.5_of_7.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/Slate/Private/Widgets/Input/SMultiLineEditableTextBox.cpp:3:
In file included from Runtime/Slate/Public/Widgets/Input/SMultiLineEditableTextBox.h:7:
In file included from Runtime/SlateCore/Public/SlateGlobals.h:7:
In file included from Runtime/SlateCore/Public/Debugging/SlateDebugging.h:8:
In file included from Runtime/SlateCore/Public/Input/Reply.h:6:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/InputCore/Classes/InputCoreTypes.h:10:10: fatal error: 'InputCoreTypes.generated.h' file not found
#include "InputCoreTypes.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
[31/73] Compile Module.SlateCore.3_of_4.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/SlateCore/Module.SlateCore.3_of_4.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Private/Rendering/RenderingPolicy.cpp:4:
In file included from Runtime/SlateCore/Public/Rendering/SlateRenderer.h:6:
In file included from Runtime/SlateCore/Public/Textures/SlateShaderResource.h:7:
In file included from Runtime/SlateCore/Public/SlateGlobals.h:7:
In file included from Runtime/SlateCore/Public/Debugging/SlateDebugging.h:8:
In file included from Runtime/SlateCore/Public/Input/Reply.h:6:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/InputCore/Classes/InputCoreTypes.h:10:10: fatal error: 'InputCoreTypes.generated.h' file not found
#include "InputCoreTypes.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[32/73] Compile Module.SSL.cpp
1 error generated.
[33/73] Compile Module.HTTP.cpp
[34/73] Compile Module.InputCore.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/InputCore/Module.InputCore.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/InputCore/Private/GenericPlatform/GenericPlatformInput.cpp:3:
In file included from Runtime/InputCore/Public/GenericPlatform/GenericPlatformInput.h:6:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/InputCore/Classes/InputCoreTypes.h:10:10: fatal error: 'InputCoreTypes.generated.h' file not found
#include "InputCoreTypes.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[35/73] Compile Module.SlateCore.2_of_4.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/SlateCore/Module.SlateCore.2_of_4.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Private/Fonts/FontFaceInterface.cpp:3:
In file included from Runtime/SlateCore/Public/Fonts/FontFaceInterface.h:8:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Public/Fonts/CompositeFont.h:9:10: fatal error: 'CompositeFont.generated.h' file not found
#include "CompositeFont.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[36/73] Compile Module.Slate.3_of_7.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/Slate/Module.Slate.3_of_7.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/Slate/Private/Framework/MultiBox/MultiBoxCustomization.cpp:3:
In file included from Runtime/Slate/Private/Framework/MultiBox/MultiBoxCustomization.h:6:
In file included from Runtime/SlateCore/Public/Widgets/DeclarativeSyntaxSupport.h:8:
In file included from Runtime/SlateCore/Public/Layout/Clipping.h:7:
In file included from Runtime/SlateCore/Public/Layout/SlateRect.h:6:
In file included from Runtime/SlateCore/Public/Layout/Margin.h:7:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Public/Types/SlateEnums.h:8:10: fatal error: 'SlateEnums.generated.h' file not found
#include "SlateEnums.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~
[37/73] Compile Module.Projects.cpp
1 error generated.
[38/73] Compile Module.SlateCore.1_of_4.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/SlateCore/Module.SlateCore.1_of_4.cpp:3:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Private/Animation/CurveSequence.cpp:4:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Public/Types/SlateEnums.h:8:10: fatal error: 'SlateEnums.generated.h' file not found
#include "SlateEnums.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~
[39/73] Compile Module.CrashDebugHelper.cpp
1 error generated.
[40/73] Compile NetCommonModule.cpp
1 error generated.
[41/73] Compile Module.SlateCore.4_of_4.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/SlateCore/Module.SlateCore.4_of_4.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Private/Widgets/SWindow.cpp:3:
In file included from Runtime/SlateCore/Public/Widgets/SWindow.h:8:
In file included from Runtime/SlateCore/Public/Layout/Margin.h:7:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Public/Types/SlateEnums.h:8:10: fatal error: 'SlateEnums.generated.h' file not found
#include "SlateEnums.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[42/73] Compile Module.Sockets.cpp
[43/73] Compile Module.Slate.7_of_7.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/Slate/Module.Slate.7_of_7.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/Slate/Private/Widgets/Text/SlateEditableTextTypes.cpp:3:
In file included from Runtime/Slate/Public/Widgets/Text/SlateEditableTextTypes.h:7:
In file included from Runtime/Slate/Public/Framework/Text/TextLayout.h:7:
In file included from Runtime/SlateCore/Public/Layout/Margin.h:7:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/SlateCore/Public/Types/SlateEnums.h:8:10: fatal error: 'SlateEnums.generated.h' file not found
#include "SlateEnums.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~
[44/73] Compile Module.ApplicationCore.cpp
[45/73] Compile Module.PakFile.cpp
[46/73] Compile Module.Slate.2_of_7.cpp
In file included from /home/carma/UnrealEngine_4.26/Engine/Intermediate/Build/Linux/B4D820EA/CrashReportClient/Shipping/Slate/Module.Slate.2_of_7.cpp:2:
In file included from /home/carma/UnrealEngine_4.26/Engine/Source/Runtime/Slate/Private/Framework/Commands/InputChord.cpp:3:
In file included from Runtime/Slate/Public/Framework/Commands/InputChord.h:7:
/home/carma/UnrealEngine_4.26/Engine/Source/Runtime/InputCore/Classes/InputCoreTypes.h:10:10: fatal error: 'InputCoreTypes.generated.h' file not found
#include "InputCoreTypes.generated.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[47/73] Compile Module.Json.cpp
1 error generated.
[48/73] Compile Module.CrashReportCore.cpp
[49/73] Compile Module.LauncherPlatform.cpp
[51/73] Compile Module.CoreUObject.4_of_10.cpp
[50/73] Compile Module.CoreUObject.3_of_10.cpp
[52/73] Compile Module.CoreUObject.1_of_10.cpp
[53/73] Compile Module.CoreUObject.8_of_10.cpp
[54/73] Compile Module.CoreUObject.9_of_10.cpp
[55/73] Compile Module.CoreUObject.5_of_10.cpp
[56/73] Compile Module.CoreUObject.6_of_10.cpp
[57/73] Compile Module.Core.12_of_18.cpp
[58/73] Compile Module.Core.3_of_18.cpp
[59/73] Compile Module.Core.6_of_18.cpp
[60/73] Compile Module.Core.8_of_18.cpp
[61/73] Compile Module.Core.11_of_18.cpp
[62/73] Compile Module.CoreUObject.7_of_10.cpp
[63/73] Compile Module.Core.2_of_18.cpp
[64/73] Compile Module.Core.10_of_18.cpp
[65/73] Compile Module.CoreUObject.10_of_10.cpp
[66/73] Compile Module.CoreUObject.2_of_10.cpp
[67/73] Compile Module.Core.16_of_18.cpp
Makefile:327: recipe for target 'CrashReportClient-Linux-Shipping' failed
make: *** [CrashReportClient-Linux-Shipping] Error 6
`
### As mentioned above, when we get to step 5, the error xxx.generation.h could not find

**Comment by pathfinder-666** [comment]
I am facing the same problem. I want to fix this. any help plz.
